proof_of_concept/definitions/asset_id.py

```diff
@@ -0,0 +1,82 @@
+"""Identifier for assets."""
+from typing import Any, cast, Type
+
+
+class AssetId(str):
+    """An ID of an asset."""
+    def __new__(cls: Type['AssetId'], seq: Any) -> 'AssetId':
+        """Create an AssetId.
+
+        Args:
+            seq: Contents, will be converted to a string using str(),
+            then used as the id.
+
+        Raises:
+            ValueError: If str(seq) is not a valid ID.
+        """
+        data = str(seq)
+        if data.startswith('id:'):
+            parts = len(data.split(':'))
+            if parts < 3 or parts > 4:
+                raise ValueError(f'Invalid asset id: {data}')
```

Comment:
```suggestion
            n_parts = len(data.split(':'))
            if n_parts not in [3, 4]:
                raise ValueError(f'Invalid asset id: {data}')
```
Maybe this is clearer?

Reply:
I like this, more Pythonic and less C++ish. I'd prefer a tuple over a list though, `(3, 4)`. I cannot edit this it seems, but maybe you can?

Reply:
Fixed by  51c2767

Reply:
Cool, yeah a tuple is nicer!